data/core.yaml

```
@@ -1766,9 +1766,9 @@ en:
       tip: "Find unroutable roads, paths, and ferry routes"
       routable:
         message:
-          one: "{highway} is disconnected from other roads and paths"
-          other: "{count} routable features are connected only to each other"
-        reference: "All roads, paths, and ferry routes should connect to form a single routing network."
+          one: "{highway} is disconnected from other commonly routable features"
+          other: "{count} commonly routable features are connected only to each other"
+        reference: "Ways consistently considered routable by data users (including roads, paths and ferry routes) should connect to form a single routing network."
```

IMHO the validation's message itself should be short and easily comprehensible. It should not use overly technical language just for the sake of covering more possible edge cases. The reference text is meant to explain what is actually going on in more detail, so that's where I would adjust this. What about something like the following?

```suggestion
          one: "{highway} is disconnected from other roads and paths"
          other: "{count} routable features are connected only to each other"
        reference: "All roads, paths, and ferry routes should connect to form a single routing network. This validation takes only features into account which are commonly and unambiguously considered to be routable."
```

---

It would be nice to focus on the selected feature(s) instead of the validation rule:

> All roads, paths, and ferry routes should connect to form a single routing network. This way is not (These ways are not) connected, or it is (they are) only connected via a feature that few applications consider routable.

---

@1ec5 sounds good. What about this:
```suggestion
          one: "{highway} is disconnected from other roads and paths"
          other: "{count} routable features are connected only to each other"
        reference: "All roads, paths, and ferry routes should connect to form a single routing network. This feature (or small collection of features) is either disconnected from the routing network or only connected to it via a feature that only few applications consider routable."
```

---

Yes, that’s good. Final nitpick is that “only few” should be either “few” or “only a few”. By the way, if you split the `reference` string into `one` and `other` strings, I think it’ll pluralize automatically.

---

Thanks for the grammar correction! :relaxed: I've amended the suggestion above.

> it’ll pluralize automatically.

I think that's only the case for the `message` part of the validation, since the `reference` part is inserted manually by the validator code, where there is currently no distinction between the `one` or `many` case: https://github.com/openstreetmap/iD/blob/v2.20.2/modules/validations/disconnected_way.js#L82-L89 – I guess that could be improved at some point, but it's not too important right now and for this PR.